### PR TITLE
Airflow orchestrator improvements

### PR DESCRIPTION
**Pull request by schustmi**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
- Airflow orchestrator doesn't start by itself now but requires a CLI call (same for shutting down)
- Added ability to monitor an orchestrator via the CLI

### CODE REVIEW

**Comment by schustmi** [comment]
This PR does not update the docs with the new Airflow Orchestrator workflow (yet). I will add it here or in a separate PR once we think the usage is how we want it to be.

File: src/zenml/orchestrators/airflow/airflow_orchestrator.py [hunk 1]
```diff
@@ -57,97 +56,164 @@ class AirflowOrchestrator(BaseOrchestrator):
...
+            )
+            destination_path = os.path.join(
+                dags_directory, os.path.basename(dag_filepath)
+            )
+            if path_utils.file_exists(destination_path):
+                logger.info(
+                    "File '{%s}' already exists, overwriting with new DAG file",
+                    destination_path,
+                )
+            shutil.copy2(dag_filepath, dags_directory)
```
Comment (htahir1) [inline src/zenml/orchestrators/airflow/airflow_orchestrator.py:147]: Maybe use `path_utils`?

File: src/zenml/orchestrators/airflow/airflow_orchestrator.py [hunk 2]
```diff
@@ -57,97 +56,164 @@ class AirflowOrchestrator(BaseOrchestrator):
...
-                        parent.kill()
-
-        # At this point, we want to reset pids
-        self.pid = None
-        self.update()
-
-        # Delete the rest
+        """Stops the airflow daemon if necessary and tears down resources."""
+        if self.is_running:
+            daemon.stop_daemon(self.pid_file, kill_children=True)
```
Comment (htahir1) [inline src/zenml/orchestrators/airflow/airflow_orchestrator.py:215]: Best line of code ever written.

File: src/zenml/utils/daemon.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+import sys
+import types
+from typing import Any, Callable, Optional
+
+import psutil
+
+from zenml.logger import get_logger
+
+logger = get_logger(__name__)
+
```
Comment (htahir1) [inline src/zenml/utils/daemon.py:13]: Please make a note that it doesnt work on Windows

**Review by htahir1 — CHANGES_REQUESTED** [review]
Great job!

File: src/zenml/cli/config.py [hunk 4]
```diff
@@ -234,29 +237,94 @@ def delete_orchestrator(orchestrator_name: str) -> None:
     cli_utils.declare(f"Deleted orchestrator: `{orchestrator_name}`")
 
 
+def _get_orchestrator(
```
[the remark this answers is not in the document]
Comment (schustmi) [reply src/zenml/cli/config.py:240]: Actually I remember why I put it here (both service and stack already have a way to get the active orchestrator or by key): The reason was that I wanted do declare something on the CLI if we fallback to the active orchestrator, so it shouldn't be part of the other classes I think. I'd argue in this case it should probably stay in the cli orchestrator file, wdyt?

**Pull request merged**